File: ios/FanZone/FanZone/Design/Components/StatusBadge.swift
import SwiftUI

// MARK: - Status Badge (Booking)

struct BookingStatusBadge: View {
    let status: BookingStatus

    var body: some View {
        HStack(spacing: 5) {
            Image(systemName: status.systemImage)
                .font(.caption2)
            Text(status.displayName)
                .font(.caption2.weight(.semibold))
        }
        .foregroundColor(status.color)
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(status.color.opacity(0.15))
        .clipShape(Capsule())
        .overlay(Capsule().stroke(status.color.opacity(0.3), lineWidth: 1))
    }
}

// MARK: - Booking Window Status Badge

struct BookingWindowBadge: View {
    let status: BookingWindowStatus

    var body: some View {
        HStack(spacing: 5) {
            Circle()
                .fill(status.color)
                .frame(width: 7, height: 7)
            Text(status.displayName)
                .font(.caption2.weight(.semibold))
                .foregroundColor(status.color)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(status.color.opacity(0.12))
        .clipShape(Capsule())
        .overlay(Capsule().stroke(status.color.opacity(0.3), lineWidth: 1))
    }
}

// MARK: - Attendance Status Badge

struct AttendanceStatusBadge: View {
    let status: AttendanceStatus

    var body: some View {
        HStack(spacing: 5) {
            Text(status.emoji)
            Text(status.displayName)
                .font(.caption.weight(.medium))
                .foregroundColor(status.color)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 4)
        .background(status.color.opacity(0.12))
        .clipShape(Capsule())
    }
}

// MARK: - Role Badge

struct RoleBadge: View {
    let role: UserRole

    private var color: Color {
        switch role {
        case .user:           return AppTheme.info
        case .bookingManager: return AppTheme.warning
        case .admin:          return AppTheme.error
        }
    }

    var body: some View {
        HStack(spacing: 4) {
            Image(systemName: role.systemImage).font(.caption2)
            Text(role.displayName).font(.caption2.weight(.semibold))
        }
        .foregroundColor(color)
        .padding(.horizontal, 8)
        .padding(.vertical, 4)
        .background(color.opacity(0.15))
        .clipShape(Capsule())
    }
}

File: flutter_app/lib/widgets/status_badge.dart
import 'package:flutter/material.dart';
import '../config/app_theme.dart';
import '../models/booking.dart';
import '../models/user.dart';

class BookingStatusBadge extends StatelessWidget {
  final BookingStatus status;

  const BookingStatusBadge({super.key, required this.status});

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
      decoration: BoxDecoration(
        color: status.color.withOpacity(0.15),
        borderRadius: BorderRadius.circular(20),
        border: Border.all(color: status.color.withOpacity(0.4)),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Icon(status.icon, size: 12, color: status.color),
          const SizedBox(width: 4),
          Text(
            status.displayName,
            style: TextStyle(
              color: status.color,
              fontSize: 11,
              fontWeight: FontWeight.w600,
            ),
          ),
        ],
      ),
    );
  }
}

class AttendanceBadge extends StatelessWidget {
  final AttendanceStatus status;

  const AttendanceBadge({super.key, required this.status});

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
      decoration: BoxDecoration(
        color: status.color.withOpacity(0.12),
        borderRadius: BorderRadius.circular(16),
      ),
      child: Text(
        status.displayName,
        style: TextStyle(
          color: status.color,
          fontSize: 11,
          fontWeight: FontWeight.w500,
        ),
      ),
    );
  }
}

class RoleBadge extends StatelessWidget {
  final UserRole role;

  const RoleBadge({super.key, required this.role});

  @override
  Widget build(BuildContext context) {
    Color color;
    IconData icon;
    switch (role) {
      case UserRole.admin:
        color = AppTheme.error;
        icon = Icons.shield;
        break;
      case UserRole.bookingManager:
        color = AppTheme.warning;
        icon = Icons.assignment_ind;
        break;
      case UserRole.user:
        color = AppTheme.info;
        icon = Icons.person;
        break;
    }

    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
      decoration: BoxDecoration(
        color: color.withOpacity(0.15),
        borderRadius: BorderRadius.circular(16),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Icon(icon, size: 12, color: color),
          const SizedBox(width: 4),
          Text(
            role.displayName,
            style: TextStyle(
              color: color,
              fontSize: 10,
              fontWeight: FontWeight.bold,
            ),
          ),
        ],
      ),
    );
  }
}

class BookingWindowBadge extends StatelessWidget {
  final bool isOpen;
  final int availableSeats;

  const BookingWindowBadge({
    super.key,
    required this.isOpen,
    required this.availableSeats,
  });

  @override
  Widget build(BuildContext context) {
    final color = !isOpen
        ? AppTheme.error
        : (availableSeats <= 10 ? AppTheme.warning : AppTheme.success);

    final text = !isOpen
        ? 'Закрыто'
        : (availableSeats <= 10 ? 'Мало мест ($availableSeats)' : 'Открыто ($availableSeats)');

    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
      decoration: BoxDecoration(
        color: color.withOpacity(0.12),
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: color.withOpacity(0.3)),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            width: 6,
            height: 6,
            decoration: BoxDecoration(color: color, shape: BoxShape.circle),
          ),
          const SizedBox(width: 5),
          Text(
            text,
            style: TextStyle(
              color: color,
              fontSize: 10,
              fontWeight: FontWeight.w600,
            ),
          ),
        ],
      ),
    );
  }
}
